npm/qsharp/src/shared/circuit.ts

```diff
@@ -55,18 +55,32 @@ export interface Qubit {
   numResults?: number;
 }
 
+export interface Measurement {
```

Comment:
You typically avoid declaring common properties twice by doing something like this:

```ts
interface SomeCommonType {
    common_prop?: string
}

export type Measurement = SomeCommonType & {
    kind: "measurement",
    measurement_prop: string
}

export type Operation = SomeCommonType & {
    kind: "operation",
    operation_prop: string
}

export type Component = Measurement | Operation
```

Reply:
And I haven't read the rest of the PR, but often, you don't even need to export or even name the `Measurement` and `Operation` subtypes - you can just write out the properties in `Component`'s definition.

Reply:
I've added a BaseOperation type to collect the shared fields, and made the Operation type a union type between the Measurement type and the Unitary type (which was the old Operation type). Let me know if this is a cleaner way to do this.